### ISSUE TITLE
.extend method with complex types validation

### ISSUE THREAD

**Issue by dwightjack** [dwightjack/vue-types#56]
Reference: https://github.com/dwightjack/vue-types/issues/54#issuecomment-510785207

We need to handle a special case for the `extend` method where we can pass to the `type` property a VueType type. Something like:

```js
VueTypes.extend({
  name: 'myType',
  getter: true,
  type: VueTypes.shape({ ... })
})
```

`VueTypes.myType` should use the object from the `shape` type as base for the new type

**Comment by dwightjack** [comment]
@micha-vb Currently I'm looking into the implementation of this feature. The branch in the is https://github.com/dwightjack/vue-types/tree/feature/56-extend-types. 

I've published a first beta version on npm (1.7.0-beta.1) under the `@next` tag. You can install it by:

```
yarn vue-types@next
```

I've also created a demo sandbox here: https://codesandbox.io/embed/vuetypes-extend-demo-v36dd

In my tests it seems to work fine, but, if you have time, it would be great if you could test it out 🙇 .

**Comment by dwightjack** [comment]
@micha-vb sorry to bother you again. Did you have the chance to test the implementation?

**Comment by micha-vb** [comment]
@dwightjack not yet. sorry. This got pushed back agian and again by my regular work. I'll try to check it out as soon as I can. But it has to wait for at least a week longer.. Hope that's ok.

**Comment by dwightjack** [comment]
@micha-vb of course! I wasn't sure you received the first message, so I thought to "ping" you again. 

It's not an urgent feature so take your time to test it.

Thanks again!

**Comment by micha-vb** [comment]
Ok, looks mostly good, but there is one case where I get this error: `TypeError: [VueTypes error]: Setting type to undefined is not allowed.`

To get that you have to nest a custom vue-type in another custom vue-type.

For example:

types.js
```
// A Vue-Router Location Object
if (!VueTypes.routerLocation) {
  VueTypes.extend({
    name: 'routerLocation',
    getter: true,
    type: VueTypes.oneOfType([
      VueTypes.shape({
        path: VueTypes.string.isRequired,
        query: VueTypes.object,
      }).loose,
      VueTypes.shape({
        name: VueTypes.string.isRequired,
        params: VueTypes.object,
      }).loose,
    ]),
  })
}

// A Vue-Router to-prop for router-links
// See: https://router.vuejs.org/api/#to
if (!VueTypes.routerTo) {
  VueTypes.extend({
    name: 'routerTo',
    getter: true,
    type: VueTypes.oneOfType([
      String,
      VueTypes.routerLocation,
    ]),
  })
}
```

Demo.vue
```
<template>
  <div>{{ url }}</div>
</template>

<script>
import VueTypes from "../types";

export default {
  props: {
    url: VueTypes.routerTo,
  }
};
</script>
```
This will work: `<Demo url="/john-doe" />`
This will throw the error: `<Demo :url="{ path: '/john-doe' }" />`

**Comment by dwightjack** [comment]
@micha-vb Thanks for your feedback!

I think the problem was in the way I was inheriting the `type` property. I've published a fix in a new beta version `1.7.0-beta.3` under the `@next` flag and updated the sandbox with your example.

Let me know if it works in your environment as well.

**Comment by micha-vb** [comment]
@dwightjack nice! The error is gone. :) Good job!

**Comment by dwightjack** [comment]
Sorry for the late reply.

Thanks for taking the time to test the release.

I've released this new feature under v1.7.0.

Closing the issue for now. Feel free to open a new issue if you eperience any problem.
